src/resolvers/git.cr

```diff
@@ -46,12 +46,16 @@ module Shards
     def initialize(@commit : String)
     end
 
+    def =~(other : GitCommitRef)
+      commit.size >= 7 && other.commit.starts_with?(commit)
```

Comment:
Why impose this minimum limit? I think git accepts a minimum of 4 digits if there is no conflicts.

Reply:
At this point it's not trivial to check whether there are no conflicts. With a minimum size of 7 the probability for a conflict is very low.

I don't think that's the correct solution though. If we keep this behaviour it should already fail when reading a hash that is too short for comparison.
But a better solution would probably to check whether the abbreviated commit is actually conflict-free by asking git to resolve it. Then we can be sure that the starts_with? match is valid.

Reply:
Because at that point it shouldn't make any assumptions at all. It would be correct even allowing no minimum size for the hash. The resolution will fail at a different place if there is not matching commit or if there is a conflict.

The second part is already being done, although it could handle the error differently to show a more meaningful message. For example, when the hash is incorrect or have conflicts it's currently showing:

```
Unable to satisfy the following requirements:

- `db (commit abcdef)` required by `shard.yml`
```

That's because `GitResolver` already returned an empty set for the available versions.